Tests/UnitTests/Mocks/PaymentComponentDelegateMock.swift

```diff
@@ -12,17 +12,37 @@ final class PaymentComponentDelegateMock: PaymentComponentDelegate {
         onDidSubmit: ((PaymentComponentData, PaymentComponent) -> Void)? = nil,
         onDidFail: ((Error, PaymentComponent) -> Void)? = nil
     ) {
-        self.onDidSubmit = onDidSubmit
-        self.onDidFail = onDidFail
+        self.onDidSubmitClosure = onDidSubmit
+        self.onDidFailClosure = onDidFail
     }
     
-    var onDidSubmit: ((PaymentComponentData, PaymentComponent) -> Void)?
+    // MARK: - onDidSubmit
+
+    var onDidSubmitCallsCount = 0
+    var onDidSubmitCalled: Bool {
+        onDidSubmitCallsCount > 0
+    }
+
+    var onDidSubmitReceivedArguments: (data: PaymentComponentData, component: PaymentComponent)?
+    var onDidSubmitClosure: ((PaymentComponentData, PaymentComponent) -> Void)?
```

Comment:
Not sure about this rename, old one is more swifty and all the other mocks still uses this style.

Reply:
Since I'm adding some extra properties in there, I believe adding "closure" makes it more explicit. I can change it if it creates more confusion.

Reply:
Yeah the `on` part achieves the closure implication. As you reverted it on the other properties only this will have `on` so closure keyword feels unnecessary imo

Reply:
Reverted